### PR TITLE
feat(experimental): `TuiAmount` add new pipe

### PR DESCRIPTION
**Pull request by vladimirpotekhin**
## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring
- [ ] Build or CI related changes
- [ ] Tests related changes
- [ ] Documentation content changes

## What is the current behaviour?

Closes #

### CODE REVIEW

**Review by splincode — APPROVED** [review]

**Review by waterplea — COMMENTED** [review]
We will eventually need to support dynamic number formatting:
https://github.com/taiga-family/taiga-ui/issues/2900
https://github.com/taiga-family/taiga-ui/issues/2578

So how about this pipe to return `Observable<string>` so the API is ready for it?

Also, setting number format through just the token is pretty inconvenient. How about we add a new `Observable` token for number formatting and make a directive for it that extends `BehaviorSubject` with `TUI_NUMBER_FORMAT` as initial value and a partial `TuiNumberFormatSettings` as input?

File: projects/experimental/pipes/amount/amount.options.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    readonly sign: TuiMoneySign;
+}
+
+export const TUI_AMOUNT_DEFAULT_OPTIONS: TuiAmountOptions = {
+    decimal: `not-zero`,
+    currency: null,
+    currencyAlign: `right`,
+    sign: `negative-only`,
+    precision: 2,
+};
```
Comment (waterplea) [inline projects/experimental/pipes/amount/amount.options.ts:20]: Should we move `decimal`, `sign` and `precision` to that new directive I proposed?

File: projects/experimental/pipes/amount/amount.module.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import {NgModule} from '@angular/core';
+import {TuiFormatNumberPipe} from '@taiga-ui/core';
+
+import {TuiAmountPipePipe} from './amount.pipe';
+
+@NgModule({
+    declarations: [TuiAmountPipePipe],
+    exports: [TuiAmountPipePipe],
+    providers: [TuiFormatNumberPipe],
```
Comment (waterplea) [inline projects/experimental/pipes/amount/amount.module.ts:9]: Looks like an unnecessary provider.

File: projects/experimental/pipes/amount/amount.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        currency = this.options.currency,
+        currencyAlign = this.options.currencyAlign,
+    ): string {
+        const sign = tuiFormatSignSymbol(value, this.options.sign);
+        const currencySymbol = tuiFormatCurrency(currency);
+        const formatted = tuiFormatNumber(Math.abs(value), {
+            ...this.format,
+            decimalLimit: this.getDecimalLimit(value),
+        });
+        // eslint-disable-next-line no-irregular-whitespace
```
Comment (waterplea) [inline projects/experimental/pipes/amount/amount.pipe.ts:31]: It should be non breaking space, otherwise it could wrap to the next line.

Comment (vladimirpotekhin) [reply projects/experimental/pipes/amount/amount.pipe.ts:31]: Yes, this exception is just for use non breaking space. Linter doesn't allow nbsp. Maybe disable this rule?

Comment (waterplea) [reply projects/experimental/pipes/amount/amount.pipe.ts:31]: Don't we have a const for it?

File: projects/core/interfaces/number-format-settings.ts [hunk 4]
```diff
@@ -28,4 +28,8 @@ export interface TuiNumberFormatSettings {
      * Enable zeros at the end of decimal part.
      */
     readonly zeroPadding: boolean;
+    /**
+     * Decimal part display mode. ('not-zero' by default)
+     */
+    readonly decimal: TuiDecimal;
```
Comment (waterplea) [inline projects/core/interfaces/number-format-settings.ts:34]: We need to make it optional until 4.0, otherwise it's a breaking change

**Review by waterplea — APPROVED** [review]

**Pull request merged**